interpreter/if_test.go

```diff
@@ -80,42 +83,117 @@ func TestInterpretIfStatement(t *testing.T) {
                }
            }
         `,
-		ParseCheckAndInterpretOptions{
-			HandleCheckerError: func(err error) {
-				errs := RequireCheckerErrors(t, err, 2)
+			ParseCheckAndInterpretOptions{
+				HandleCheckerError: func(err error) {
+					errs := RequireCheckerErrors(t, err, 2)
 
-				assert.IsType(t, &sema.UnreachableStatementError{}, errs[0])
-				assert.IsType(t, &sema.UnreachableStatementError{}, errs[1])
+					assert.IsType(t, &sema.UnreachableStatementError{}, errs[0])
+					assert.IsType(t, &sema.UnreachableStatementError{}, errs[1])
+				},
 			},
-		},
-	)
-	require.NoError(t, err)
-
-	for name, expected := range map[string]int64{
-		"testTrue":   2,
-		"testFalse":  3,
-		"testNoElse": 2,
-		"testElseIf": 3,
-	} {
-		t.Run(name, func(t *testing.T) {
-			value, err := inter.Invoke(name)
-			require.NoError(t, err)
-
-			AssertValuesEqual(
-				t,
-				inter,
-				interpreter.NewUnmeteredIntValueFromInt64(expected),
-				value,
-			)
-		})
-	}
+		)
+		require.NoError(t, err)
+
+		for name, expected := range map[string]int64{
+			"testTrue":   2,
+			"testFalse":  3,
+			"testNoElse": 2,
+			"testElseIf": 3,
+		} {
+			t.Run(name, func(t *testing.T) {
+				value, err := inter.Invoke(name)
+				require.NoError(t, err)
+
+				AssertValuesEqual(
+					t,
+					inter,
+					interpreter.NewUnmeteredIntValueFromInt64(expected),
+					value,
+				)
+			})
+		}
+	})
+
+	// Note: compiler can't run programs with unreachable-statement errors
+	// (i.e: when type checking is skipped for some part of the code),
+	// because the compiler relies on the type information produced by the checker.
```

Comment:
We should probably mark unreachable code in the checker and not even compile it

Reply:
Yeah we could do that too. I guess that would only be needed for tests, and wont be used in real networks (?). Added as an item for https://github.com/onflow/cadence/issues/3804